File: 初级算法/动态规划/买卖股票的最佳时机.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int buy = INT_MAX;
        int sell = 0;
        vector<int>::iterator pos = prices.begin();
        while(pos != prices.end()){
            buy =  min(buy, *pos);              //不断替换最小值       //min函数取两个参数中的小值,max函数取两个参数中的大值
            sell = max(sell, *pos - buy);       //sell值不断替换为更大利润的sell值
            ++ pos;                             //单向移动pos指针，不会出现卖出时间在买入时间之前的情况
        }
        return sell;
        
    }
};

File: java/买卖股票的最佳时机.java
class Solution {
    public int maxProfit(int[] prices) {
        int profit = 0;
        int tmp = 0;
        if(prices.length == 0)
            return 0;
        for(int i = 0;i < prices.length;i ++)
            for(int j = i;j < prices.length;j ++){
                tmp = prices[j] - prices[i];
                if( tmp > profit)
                    profit = tmp;
                else
                    continue;
            }
        return profit;
    }    
}